crates/e2e/src/setup/proxy.rs

```diff
@@ -0,0 +1,117 @@
+use {
```

Comment:
Pls add a comprehensive comment why this exist: to simulate a k8s `service` object that load balances over multiple instances.